sharding-orchestration/sharding-orchestration-core/src/main/java/io/shardingsphere/orchestration/internal/registry/config/listener/SchemaChangedListener.java

```diff
@@ -0,0 +1,138 @@
+        if (event.getKey().equals(configurationNode.getDataSourcePath(shardingSchemaName))) {
+            return createDataSourceChangedEvent(shardingSchemaName, event);
+        }
+        return createRuleChangedEvent(shardingSchemaName, event);
+    }
+    
+    private DataSourceChangedEvent createDataSourceChangedEvent(final String shardingSchemaName, final DataChangedEvent event) {
+        return new DataSourceChangedEvent(shardingSchemaName, ConfigurationYamlConverter.loadDataSourceConfigurations(event.getValue()));
+    }
+    
+    private ShardingOrchestrationEvent createRuleChangedEvent(final String shardingSchemaName, final DataChangedEvent event) {
+        return isShardingRule(event) ? createShardingRuleChangedEvent(shardingSchemaName, event.getValue()) : createMasterSlaveRuleChangedEvent(shardingSchemaName, event.getValue());
+    }
+    
+    private boolean isShardingRule(final DataChangedEvent event) {
+        return event.getValue().contains("tables:\n");
+    }
+    
+    private ShardingRuleChangedEvent createShardingRuleChangedEvent(final String shardingSchemaName, final String ruleValue) {
+        return new ShardingRuleChangedEvent(shardingSchemaName, ConfigurationYamlConverter.loadShardingRuleConfiguration(ruleValue));
+    }
+    
+    private MasterSlaveRuleChangedEvent createMasterSlaveRuleChangedEvent(final String shardingSchemaName, final String ruleValue) {
+        return new MasterSlaveRuleChangedEvent(shardingSchemaName, ConfigurationYamlConverter.loadMasterSlaveRuleConfiguration(ruleValue));
+    }
+    
+    private ShardingOrchestrationEvent createChangedEventForNewSchema(final String shardingSchemaName) {
+        if (isSufficientToInitialize(shardingSchemaName)) {
+            existedSchemaNames.add(shardingSchemaName);
+            return createSchemaChangedEvent(shardingSchemaName);
+        }
+        return new IgnoredChangedEvent();
+    }
+    
+    private boolean isSufficientToInitialize(final String shardingSchemaName) {
+        return configurationService.hasDataSourceConfiguration(shardingSchemaName) && configurationService.hasRuleConfiguration(shardingSchemaName);
+    }
+    
+    private SchemaAddChangedEvent createSchemaChangedEvent(final String shardingSchemaName) {
+        return new SchemaAddChangedEvent(shardingSchemaName, configurationService.loadDataSourceConfigurations(shardingSchemaName), createRuleConfiguration(shardingSchemaName));
+    }
+    
+    private RuleConfiguration createRuleConfiguration(final String shardingSchemaName) {
+        return configurationService.isShardingRule(shardingSchemaName) 
+                ? configurationService.loadShardingRuleConfiguration(shardingSchemaName) : configurationService.loadMasterSlaveRuleConfiguration(shardingSchemaName);
+    }
+    
+    private ShardingOrchestrationEvent createDeleteChangedEvent(final String shardingSchemaName) {
```

Comment:
How about rename this method as `createDeletedEvent`?

Reply:
Please refer other event names.